packages/core/src/declareAtom.ts

```diff
@@ -120,8 +121,18 @@ export function declareAtom<TState>(
             `Invalid state. Reducer number ${position} in "${_name}" atom returns undefined`,
           )
 
-        if (atomStateNew !== atomState && !hasAtomNewState) changedIds.push(_id)
-        stateNew[_id as string] = atomStateNew
+        if (hasAtomNewState && equals(atomStateSnapshot, atomStateNew)) {
+          changedIds.splice(changedIds.indexOf(_id), 1)
+          delete stateNew[_id as string]
```

Comment:
May be it is unnecessary? Subscribers calls by changedIds and assign equal newState to state is do nothing?

Reply:
May be, but then the dependency will be marked changed for the subscribers.
https://github.com/artalar/reatom/blob/dc1b26b214f8bf34f8fb9b5fdb53196c5702889e/packages/core/src/declareAtom.ts#L111-L112